Source/WebCore/html/HTMLAttachmentElement.h

```diff
@@ -61,7 +60,9 @@ class HTMLAttachmentElement final : public HTMLElement {
     WEBCORE_EXPORT void updateAttributes(std::optional<uint64_t>&& newFileSize, const AtomString& newContentType, const AtomString& newFilename);
     WEBCORE_EXPORT void updateEnclosingImageWithData(const String& contentType, Ref<FragmentedSharedBuffer>&& data);
     WEBCORE_EXPORT void updateThumbnail(const RefPtr<Image>& thumbnail);
+    WEBCORE_EXPORT void updateThumbnailImgSrcData(Vector<uint8_t>&&);
     WEBCORE_EXPORT void updateIcon(const RefPtr<Image>& icon, const WebCore::FloatSize&);
+    WEBCORE_EXPORT void updateIconImgSrcData(Vector<uint8_t>&&);
```

Comment:
Ditto for naming here, I think we should just use `updateThumbnail` & `updateIcon` as function names, and change the old ones to be `updateLegacyThumbnail/Icon`.

Reply:
I think the naming is valuable to keep here: The argument is just a bag of bytes, so it's good to know that it's actually intended as a data blob for the `<img>`'s `src`.
Also, as above, the "old" UI is still the default.

Reply:
I mean, `updateThumbnailImgSrcData` and `updateIconImgSrcData` end up calling `updateImage`, so from the user perspective, you are just updating the thumbnail and the icon.

Reply:
`updateImage` is private.

Reply:
I guess my point is that these functions call `updateImage`, so they're not just updating the srcData fields, they're also updating the actual displayed image.

Reply:
These functions provide the thumbnail/icon data for a `src` field, then attachment *may* update the displayed image with that data.

Anyway, more importantly here I think: `WebPage` does a `if (wide) updateImgSrcData else update` and I've added assertions to verify that, so I now think that the names should reflect that precondition. Also I'm removing mentions of "legacy". So I'll change these functions to `Update{Narrow,Wide}{Thumbnail,Icon}`.